### PR TITLE
RHOARDOC-1645: [cherry-pick] remove references to specific maven an JDK versions

### PR DESCRIPTION
**Pull request by inoxx03**
- [ ] Have you followed the guidelines in our [Contributing](https://thorntail.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [ ] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by tradej** [comment]
ok to test

**Comment by Ladicek** [comment]
ok to test

**Review by tradej — DISMISSED** [review]
LGTM.

File: docs/howto/writing-an-application-from-scratch/index.adoc [hunk 1]
```diff
@@ -7,8 +7,8 @@ Creating a simple {Thorntail}{ndash}based application with a REST endpoint from
 [discrete]
 == Prerequisites
 
-* JDK 8 or newer installed
-* Maven 3.3.x or newer installed
+* JDK installed
```
Comment (Ladicek) [inline docs/howto/writing-an-application-from-scratch/index.adoc]:
```suggestion
* JDK 8 or 11 installed
```

File: docs/howto/writing-an-application-from-scratch/index.adoc [hunk 2]
```diff
@@ -7,8 +7,8 @@ Creating a simple {Thorntail}{ndash}based application with a REST endpoint from
...
+* Maven installed
```
Comment (Ladicek) [inline docs/howto/writing-an-application-from-scratch/index.adoc]:
```suggestion
* Maven 3.5.0 or newer installed
```

**Comment by Ladicek** [comment]
So, frankly, I'm not sure what to think about this. The difference between Thorntail docs and the RHOAR Thorntail release notes is minor, and the suggestion to just leave out versions from Thorntail docs doesn't really remove the difference. It just muddies the water. I added suggestions that would IMHO make it better and consistent with the release notes.

**Comment by tradej** [comment]
The rationale was to reduce the maintenance burden by moving the version requirement only to the Release Notes, but if you believe this reduction goes too far, we'll keep it.

**Comment by Ladicek** [comment]
I understand the desire, but I don't want to treat the community project as a second-class citizen ;-)

**Review by tradej — APPROVED** [review]

**Review by Ladicek — APPROVED** [review]
Well, OK...

**Comment by tradej** [comment]
@Ladicek I totally understand and agree, I'm probably just over-eager in adapting to the low staffing and high turnover.

**Pull request merged**